test/Microsoft.ML.Tests/DatabaseLoaderTests.cs

```diff
@@ -37,43 +39,84 @@ public void IrisLightGbm()
                 new DatabaseLoader.Column() { Name = "PetalWidth", Type = DbType.Single }
             };
 
-            using (var connection = new MockConnection(mlContext, GetDataPath(TestDatasets.iris.trainFilename), loaderColumns))
+            var loader = mlContext.Data.CreateDatabaseLoader(loaderColumns);
+
+            var trainingData = loader.Load(() => new MockDatabaseReader(mlContext, loaderColumns, connectionString, commandText));
```

Comment:
I'm okay if we have units tests using a mock, but we should also have a real **integration test** using at least **SQL locadb**. Sometimes using mocks might have slightly differences that can impact us..

Reply:
Yes, I agree 100%.

However, I don't believe we have anything setup for doing this in any of the repos, and I am not checking a database into git, so we need a solution for pulling that down and managing it (even if its an azure blob).

Reply:
For the the .mdf and .ldf files we could have the same approach to what we do for test dataset files. We have NuGet packages with it since they are small datasets just for tests. We can probably do the same for the localdb files?

For tests, the database files should be kind of small. If testing with large databases those kind of tests shouldn't be running on localdb and our Azure Pipelines unless the database is published in an standalone Azure SQL Database so no large file is downloaded when running the tests.

I also have very large databases (up to 50GB with more than 100 million records) ready to be used for testing in an Azure SQL DB. But this is more for end-to-end sample apps testing.